compiler/rustc_mir_build/src/build/expr/as_temp.rs

```diff
@@ -70,7 +70,7 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                     local_decl.local_info =
                         Some(Box::new(LocalInfo::StaticRef { def_id, is_thread_local: true }));
                 }
-                ExprKind::Literal { const_id: Some(def_id), .. } => {
+                ExprKind::NamedConst { def_id, .. } => {
```

Comment:
This might be problematic. I'm not completely sure whether we need any other variants here. Also don't know what adding constants to local_info here does later in codegen.

Reply:
we previously also did this for `ExprKind::ConstParam`, would have to figure out what it's used for myself :laughing: think you can just keep the current behavior and add a FIXME for now
```suggestion
                ExprKind::NamedConst { def_id, .. } | ExprKind::ConstParam { def_id, .. } => {
```

Reply:
we also want this for `ExprKind::ConstParam`: please add this as a test:
```rust
#![feature(adt_const_params)]
#![allow(incomplete_features)]

#[derive(PartialEq, Eq)]
struct Yikes;

impl Yikes {
    fn mut_self(&mut self) {}
}

fn foo<const YIKES: Yikes>() {
    YIKES.mut_self()
}

fn main() {
    foo::<{ Yikes }>()
}
```
https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=fd11458a9e567c54fbb2fb43c15c9f56